pages/tokens/user_activation.php

```diff
@@ -1,12 +1,60 @@
 <?php
 $token = AccessToken::retrieve($_GET['token'] ?? "");
-page("Activer le compte") ?>
+page("Activer le compte");
+$v = new Validator(action: "validate_form");
+$new_password = $v->password("new_password")->autocomplete("new-password")->placeholder("Nouveau mot de passe")->required()->secure();
+$confirm_password = $v->password("confirm_password")
+    ->autocomplete("new-password")
+    ->placeholder("Confirmer le mot de passe")
+    ->required();
+$confirm_password->condition($new_password->value == $confirm_password->value, "Les deux mots de passe sont différents");
+$gender = $v->text("gender")->label("Sexe");
+$phone = $v->phone("phone")->label("Numéro de téléphone")->placeholder();
+
+if ($v->valid()) {
+    $user = $token->user;
+    $user->set_password($new_password->value);
+    $user->gender = Gender::from($gender->value);
+    $user->phone = $phone->value;
+    $user->active = true;
+    em()->persist($user);
+    em()->flush();
+    $v->set_success("Mot de passe mis à jour !");
+    redirect("/");
```

Comment:
Because of this redirection, the success message will never display.

We could implement a flash message mechanism to allow messages to be displayed beyond redirections